[Exam] Enhancement/exam/exercise layout

fde312 opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
Separates the problem statement from the students input. 
Closes #1589 #1588 #1587 

### Description
<!-- Describe your changes in detail -->

- Use `resizable-container.component` to separate the problem statement and make it resizable/collapsable for Text, Modeling and File Upload Exercises
- Use `media (max-width: 991.98px)` for smaller screens
- Removed border and header for exercise submissions
- Small changes for team exercises and modeling exercise

### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->

1. Log in to Artemis
2. Navigate to Course Administration
3. Create Text, Modeling, FileUpload Exercises and set start date to now and due date in a few minutes
4. Participate in Exercises and check if the Problem Statement is resizable/collapsable
4.1 Check if for smaller screens the Problem Statement aligns above the students input and is not resizable/collapsable
5. After Due Date see your Submission and check if Problem Statement is resizable/collapsable
5.1 Check if for smaller screens the Problem Statement aligns above the students input and is not resizable/collapsable
6. Check also for team exercises

Note: `resizable-container.component` is also used in `text-submission-assessment` 
7. Check if assessment instructions align above students submission for smaller screens

### Screenshots
<!-- Add screenshots to demonstrate the changes in the UI. -->
<!-- Create a GIF file from a screen recording in a docker container https://toub.es/2017/09/11/high-quality-gif-with-ffmpeg-and-docker/ -->
File Upload:
<img width="1277" alt="image" src="https://user-images.githubusercontent.com/41108487/85124216-d4b11600-b229-11ea-963f-7008623cfb16.png">


Modeling:
<img width="1271" alt="image" src="https://user-images.githubusercontent.com/41108487/85124021-869c1280-b229-11ea-80af-e52225d2b192.png">



Text:
<img width="1268" alt="image" src="https://user-images.githubusercontent.com/41108487/85124153-b814de00-b229-11ea-8b6f-bbf5191cf5c1.png">


Smaller devices:
<img width="766" alt="image" src="https://user-images.githubusercontent.com/41108487/85124109-a7fcfe80-b229-11ea-8104-b993aa244992.png">

<img width="779" alt="image" src="https://user-images.githubusercontent.com/41108487/85124178-c3680980-b229-11ea-99f6-4fe9603b853f.png">

<img width="770" alt="image" src="https://user-images.githubusercontent.com/41108487/85124252-e2669b80-b229-11ea-8e02-4d74bb29d72d.png">




Team exercises:
<img width="1239" alt="image" src="https://user-images.githubusercontent.com/41108487/85123912-5e141880-b229-11ea-905f-18f79e69f9c9.png">

<img width="1245" alt="image" src="https://user-images.githubusercontent.com/41108487/85123937-69674400-b229-11ea-8d0e-047ce4cc541a.png">

Team exercises on smaller devices:
<img width="759" alt="image" src="https://user-images.githubusercontent.com/41108487/85124307-fa3e1f80-b229-11ea-9dca-a416dfbdf32d.png">

<img width="747" alt="image" src="https://user-images.githubusercontent.com/41108487/85124349-104be000-b22a-11ea-8759-6a85bc3974b4.png">

Alexanderkrp commented:
Looks good to me 👍 
One thought: 
For a wide screen there is this threshold about 40 percent of the right side screen which the problem statement cannot surpass, wouldn't it be good if also the PC users were able to have the problem statement above when it is expanded more or that the textbox is detachable so that students/ tutors(for grading criteria) can drag it whereever they like it best?

fde312 commented:
> Mostly everything worked just fine for me, good job! 💯
> One issue I noticed about the modeling exercise full screen option is the following:
> ![gitFDE](https://user-images.githubusercontent.com/50466283/84886043-29bb2380-b094-11ea-8ab1-12ab4194155a.gif)
> (One is not able to see the "dragging" of the components in full screen mode anymore)

I talked to @TobiasPr and he said this bug was fixed with the last merge to develop. So this is already fixed! :)

schultek commented on src/main/webapp/app/shared/resizeable-container/resizeable-container.component.scss:
```diff
@@ -97,3 +98,28 @@
+    }
+
+    .expanded {
+        width: 100%;
```
```suggestion
        width: 100% !important;
```

I found a small issue: When dragging the resizable container to some width and then resizing the window to a smaller width so that the containers are displayed on top of each other, the previous resized container stays at its width and does not expand to full width. This is because the inline css style overrides the media query style.
<img width="890" alt="Bildschirmfoto 2020-06-18 um 11 11 06" src="https://user-images.githubusercontent.com/13920539/85002079-c6dd9100-b154-11ea-84d2-f480f097b5d3.png">
<img width="548" alt="Bildschirmfoto 2020-06-18 um 11 11 29" src="https://user-images.githubusercontent.com/13920539/85002101-cb09ae80-b154-11ea-8a05-c5d940ff1729.png">

Adding an !important to the media query fixes this.

TobiasPr commented:
Two small things I found is that the resize container is "jumping" when I start to resize. So it 
![jumping_resize_container](https://user-images.githubusercontent.com/22955066/84998395-0f468000-b150-11ea-82f4-6dc3014350a4.gif)

Another small styling issue:
![image](https://user-images.githubusercontent.com/22955066/84998669-6c423600-b150-11ea-89f3-b1508260d1cd.png)

Other than that, it worked really good for me. Good job!

krusche commented:
I don't really like that the exercise submission is wrapped in another box that shows the title (which is already shown in the header), so it's basically duplicated information that is not really necessary.

Can we leave out the left box and implement it like the exercise details page, i.e. only a box on the right side (in this case question and answer) and no box on the left side?

![image](https://user-images.githubusercontent.com/744067/85014448-93a3fd80-b166-11ea-8b8b-71d30514fda9.png)

fde312 commented:
> Looks good to me 👍
> One thought:
> For a wide screen there is this threshold about 40 percent of the right side screen which the problem statement cannot surpass, wouldn't it be good if also the PC users were able to have the problem statement above when it is expanded more or that the textbox is detachable so that students/ tutors(for grading criteria) can drag it whereever they like it best?

@Alexanderkrp Maybe you can write a new issue for this and then we can think of it later on. Right now I would say that this is the UI that we would like to have for the exam mode :)

fde312 commented:
@krusche I changed the left component as discussed and adjusted the view for team exercises and modeling exercise.

I would like to fix the problem with the jumping in a follow-up PR if this is ok :)

TobiasPr commented on src/main/webapp/app/exercises/text/participate/text-editor.component.scss:
```diff
@@ -20,8 +20,8 @@
     .grid-area-team-info {
-        grid-area: team-info;
-        justify-self: flex-end;
+        //grid-area: team-info;
```
can we remove that?

fde312 replied:
removed :)

fde312 commented:
> Here is a small thing that I found. On smaller screens there are multiple scroll bars for modeling exercises. The one from the modeling editor (wanted) and the second one from the resizable-container.component. Would be nicer if the editor would fit inside and the scrolling bar of the resizable component would not be displayed.
> ![image](https://user-images.githubusercontent.com/22955066/85169097-92a8c400-b26b-11ea-819a-5aea8d3814ba.png)

@TobiasPr the second scrollbar should be fixed as well as the jumping when resizing :) Would be great if you could re-test :)

krusche commented:
@fde312 what is the status of this PR? I think we should get this done rather quickly.

fde312 commented:
> @fde312 what is the status of this PR? I think we should get this done rather quickly.

@krusche I could not reproduce the errors mentioned by @balazs-czopf. We will re-test this together and see if we can reproduce them.

balazs-czopf commented:
After testing together with @fde312 we found the bug responsible for previously mentioned screenshots one and three. The other bugs are not part of this PR and were put in [Issue 1697](https://github.com/ls1intum/Artemis/issues/1697).
